extensions/incubator/src/main/java/io/opentelemetry/extension/incubator/propagation/PassThroughPropagator.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.extension.incubator.propagation;
+
+import static java.util.Objects.requireNonNull;
+
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.ContextKey;
+import io.opentelemetry.context.propagation.ContextPropagators;
+import io.opentelemetry.context.propagation.TextMapGetter;
+import io.opentelemetry.context.propagation.TextMapPropagator;
+import io.opentelemetry.context.propagation.TextMapSetter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+import javax.annotation.Nullable;
+
+/**
+ * A {@link TextMapPropagator} which can be configured with a set of fields, which will be extracted
+ * and stored in {@link Context}. If the {@link Context} is used again to inject, the values will be
+ * injected as-is. This {@link TextMapPropagator} is appropriate for a service that does not need to
+ * participate in telemetry in any way and provides the most efficient way of propagating incoming
+ * context to outgoing requests. In almost all cases, you will configure this single {@link
+ * TextMapPropagator} when using {@link
+ * io.opentelemetry.api.OpenTelemetry#propagating(ContextPropagators)} to create an {@link
+ * io.opentelemetry.api.OpenTelemetry} that only propagates. Similarly, you will never need this
+ * when using the OpenTelemetry SDK to enable telemetry.
+ */
+public final class PassThroughPropagator implements TextMapPropagator {
+
+  private static final ContextKey<List<String>> EXTRACTED_KEY_VALUES =
+      ContextKey.named("passthroughpropagator-keyvalues");
+
+  /**
+   * Returns a {@link TextMapPropagator} which will propagate the given {@code fields} from
+   * extraction to injection.
+   */
+  public static TextMapPropagator create(String... fields) {
+    requireNonNull(fields, "fields");
+    return create(Arrays.asList(fields));
+  }
+
+  /**
+   * Returns a {@link TextMapPropagator} which will propagate the given {@code fields} from
+   * extraction to injection.
+   */
+  public static TextMapPropagator create(Iterable<String> fields) {
+    requireNonNull(fields, "fields");
+    List<String> fieldsList =
+        StreamSupport.stream(fields.spliterator(), false)
+            .map(field -> requireNonNull(field, "field"))
+            .collect(Collectors.toList());
+    if (fieldsList.isEmpty()) {
+      return TextMapPropagator.noop();
+    }
+    return new PassThroughPropagator(fieldsList);
+  }
+
+  private final List<String> fields;
```

Comment:
I thought we agreed this goes up top, not below the static methods. 😞 
It's even in the CONTRIBUTING.md :)

Reply:
Ah I think it was less agree and more just outvote :P Think I'll continue to forget this in the future